src/main.py

```diff
@@ -1101,6 +1101,7 @@ def _simple_primary_switch(self, limit, new_primary, is_dead):
         logging.info('Starting simple primary switch to {}'.format(new_primary))
         if self.checks['primary_switch'] >= primary_switch_checks:
             self._set_simple_primary_switch_try()
+        self.db.enable_wal_receiver()
```

Comment:
Looks like a possible data loss.
Just after failover and loosing election, this host still have old master in recovery.conf.
`enable_wal_receiver` will make the replica to continue streaming from old master.
It may confirm more LSNs, and they will be lost.

We can enable wal receiver only after updating recovery.conf